packages/@aws-cdk/aws-servicecatalog/lib/product.ts

```diff
@@ -0,0 +1,212 @@
+import { ArnFormat, IResource, Resource, Stack } from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { CloudFormationTemplate } from './cloudformation-template';
+import { AcceptLanguage } from './common';
+import { InputValidator } from './private/validation';
+import { CfnCloudFormationProduct } from './servicecatalog.generated';
+
+/**
+ * A Service Catalog product, currently only supports type CloudFormationProduct
+ */
+export interface IProduct extends IResource {
+  /**
+   * The ARN of the product.
+   * @attribute
+   */
+  readonly productArn: string;
+
+  /**
+   * The id of the product
+   * @attribute
+   */
+  readonly productId: string;
+}
+
+abstract class ProductBase extends Resource implements IProduct {
+  public abstract readonly productArn: string;
+  public abstract readonly productId: string;
+}
+
+/**
+ * Properties of product version (also known as a provisioning artifact).
+ */
+export interface ProductVersions {
```

Comment:
Sorry, I think there's been some misunderstanding.

The name of this interface should be `CloudFormationProductVersion`.

The name of the property is a different discussion.

Reply:
ok I think i fixed it Adam, is this what you mean?

Reply:
Yes.